### ISSUE TITLE
getPackage API doesn't work from R on Windows

### ISSUE THREAD

**Issue by amoeba** [NCEAS/metacat#1474]
Downloading packages via the `getPackage` API via R results in a ZIP file that can't be extracted either from R or using other tools once downloaded via R. AFAICT this is just a Windows+R thing. Chrome on Windows downloads these files fine. For example:

```r
pkg_url <- "https://knb.ecoinformatics.org/knb/d1/mn/v2/packages/application%2Fbagit-097/resource_map_urn%3Auuid%3A0bc6c1fd-2070-49e8-9900-a6fd5e4c2f95"
download.file(pkg_url, "pkg.zip", method = "libcurl")
unzip("pkg.zip", unzip = "unzip")
#Warning message:
#  In unzip(zipfile = "pkg.zip", exdir = getwd()) :
#  error 1 in extracting from zip file
```

A hex dump shows the files are subtly different when you download from R (as above) vs. via a web browser (even on Windows). I'm pretty sure we use stock Java ZIP functionality so I expect that's not the culprit here. I wonder if this has to do with the value we set in the `Content-Disposition` header, which is:

> `'inline; filename="resource_map_urn_uuid_0bc6c1fd_2070_49e8_9900_a6fd5e4c2f95.zip"'`

Note the use of `inline` whereas it would be more natural to use `attachment` to indicate to the client the response is something that should be saved to the filesystem.

**Comment by mbjones** [comment]
Good catch, @amoeba Do the bytes of the payload checksum match in the on-the-wire serialization (e.g., before the MIME payload is extracted and saved to disk) in the R versus Chrome scenarios? It would be interesting to see if curl with spoofed user agent headers for chrome and R produces checksum identical outputs from the server, and whether the resulting files when extracted from the MIME wrappers and serialized can be read with various tools. If the server is producing different outputs, that would be strange. And if not, then we should be able to figure out what's going on in the serialize to disk client-side process that differs.

**Comment by amoeba** [comment]
This _might_ not be anything to do with metacat after all. I decided to try `curl::curl_download` just to verify and that method produces a valid ZIP on Windows which can be extracted via `utils::unzip` and external ZIP applications. And `download.file` with `method = "curl"` instead of `libcurl` works too.

I took a dive into the `download.file` docs and it looks like I can explain why the `libcurl` method doesn't work: It doesn't do binary transfer if the URL doesn't end in

> one of .gz, .bz2, .xz, .tgz, .zip, .rda, .rds

Manually specifying the mode like `download.file(pkg_url, "pkg.zip", method = "libcurl", mode = "wb")` makes a call using the `libcurl` method work.

So it looks like this isn't really something to fix in Metacat after all and that it's just something to keep in mind when we write docs and do trainings:

Windows users should use `download.file` with either the `libcurl` method and set `mode = "wb"` or they should use the `curl` method. Alternatively, `curl::curl_download`. I don't think I want to see us put `.zip` in the URL. We already set `application/zip` as the `Content-Type` and I think R is just doing the wrong thing here.
